scripts/netContain/scripts/contivNet.sh

```diff
@@ -29,126 +18,94 @@ touch /tmp/restart_netplugin
 
 #Needed for netplugin to connect with OVS, This needs to be
 #fixed as well. netplugin should have OVS locally.
-echo "0.0.0.0 localhost" >>/etc/hosts
-
-while getopts ":xmp:v:i:c:drl:o:t:" opt; do
-	case $opt in
-		m)
-			netmaster=true
-			netplugin=false
-			;;
-		v)
-			vtep_ip=$OPTARG
-			netplugin=true
-			;;
-		i)
-			vlan_if=$OPTARG
-			netplugin=true
-			;;
-		c)
-			cstore=$OPTARG
-			;;
-		p)
-			plugin=$OPTARG
-			;;
-		r)
-			cleanup=true
-			;;
-		x)
-			reinit=true
-			;;
-		d)
-			debug="--log-level debug"
-			;;
-		l)
-			listen_url=$OPTARG
-			;;
-		o)
-			control_url=$OPTARG
-			;;
-		:)
-			echo "An argument required for $OPTARG was not passed"
-			;;
-		?)
-			echo "Invalid option supplied"
-			;;
-	esac
-done
-
-if [ $cleanup == false ] && [ $netplugin == true ]; then
-	echo "Initializing OVS"
-	/contiv/scripts/ovsInit.sh
-	echo "Initialized OVS"
-fi
-
-if [ $cleanup == true ] || [ $reinit == true ]; then
-	ovs-vsctl del-br contivVlanBridge || true
-	ovs-vsctl del-br contivVxlanBridge || true
-	for p in $(ifconfig | grep vport | awk '{print $1}'); do
-		ip link delete $p type veth
-	done
-	rm -f /opt/cni/bin/contivk8s || true
-	rm -f /etc/cni/net.d/1-contiv.conf || true
+echo "0.0.0.0 localhost" >> /etc/hosts
+
+if [ -z "$CONTIV_ROLE" ]; then
+    echo "CRITICAL: ENV CONTIV_ROLE must be set"
+    echo "CRITICAL: Unknown contiv role"
+    exit 1
+elif [ "$CONTIV_ROLE" != "netmaster" ] && [ "$CONTIV_ROLE" != "netplugin" ]; then
+    echo "CRITICAL: ENV CONTIV_ROLE must be in [netmaster, netplugin]"
+    echo "CRITICAL: Unknown contiv role"
+    exit 1
 fi
+echo "INFO: Running contiv as $CONTIV_ROLE"
 
-if [ $cleanup == true ]; then
-	exit 0
+if [ ! -z "$CONTIV_MODE" ]; then
+    if [ "$CONTIV_ROLE" = "netmaster" ] && [ -z "$CONTIV_NETMASTER_MODE" ] ; then
+        CONTIV_NETMASTER_MODE="$CONTIV_ROLE"
+    elif [ "$CONTIV_ROLE" = "netplugin" ] && [ -z "$CONTIV_NETPLUGIN_MODE" ] ; then
+        CONTIV_NETPLUGIN_MODE="$CONTIV_ROLE"
+    fi
+elif [ ! -z "$CONTIV_NETMASTER_MODE" ]; then
+    CONTIV_MODE="$CONTIV_NETMASTER_MODE"
+elif [ ! -z "$CONTIV_NETPLUGIN_MODE" ]; then
+    CONTIV_MODE="$CONTIV_NETPLUGIN_MODE"
+else
+    echo "CRITICAL: ENV CONTIV_MODE or CONTIV_NETMASTER_MODE or CONTIV_NETPLUGIN_MODE must be set"
+    echo "CRITICAL: Unknown contiv mode"
+    exit 1
 fi
+echo "INFO: Running contiv in mode $CONTIV_MODE"
 
-if [ $netplugin == false ] && [ $netmaster == false ]; then
-	echo "No netmaster or netplugin options were specified"
-	exit 1
-fi
+set -uo pipefail
 
 mkdir -p /opt/contiv/
-mkdir -p /var/contiv/log/
 
-if [ "$plugin" == "kubernetes" ]; then
-   mkdir -p /opt/contiv/config
-   mkdir -p /var/contiv/config
-   echo ${CONTIV_CONFIG} >/var/contiv/config/contiv.json
-   cp /var/contiv/config/contiv.json /opt/contiv/config/contiv.json
+if [ -d /var/contiv/log ]; then
```

Comment:
Could both of these dirs even even exist at the same time?  I think that implies both bindmounts are active... and in that case, the `mv` here is moving a bindmount onto another bindmount (gonna fail)

I think moving the _contents_ will work, but moving the dir (possibly mountpoint) is not safe

Reply:
Ah yeah I see these are both bindmounts... you'll need to `mv` the contents (or `cp -r`), not the directory (which is actually a mountpoint)

Reply:
changed to use `cp -a`